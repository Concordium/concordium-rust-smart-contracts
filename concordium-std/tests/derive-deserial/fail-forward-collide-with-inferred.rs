//! Ensure `derive(Deserial)` fails when a 'forward' attribute collides with the
//! default tag for another variant.
use concordium_std::*;

#[derive(Deserial)]
#[concordium(repr(u8))]
enum Count {
    One {
        field: u32,
    },
    #[concordium(forward = [0, 6])]
    Two(Inner),
}

#[derive(Deserial)]
#[concordium(repr(u8))]
enum Inner {
    Alpha {
        balance: u32,
    },
    #[concordium(tag = 6)]
    Beta(u16),
}

fn main() {}
