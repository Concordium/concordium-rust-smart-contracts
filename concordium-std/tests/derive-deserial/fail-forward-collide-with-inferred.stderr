error: The tag of 'Two' collides with the tag of 'One'.
  --> tests/derive-deserial/fail-forward-collide-with-inferred.rs:11:29
   |
11 |     #[concordium(forward = [0, 6])]
   |                             ^

error: The tag of 'One' collides with the tag of 'Two'.
 --> tests/derive-deserial/fail-forward-collide-with-inferred.rs:8:5
  |
8 |     One {
  |     ^^^
